scripts/functional-tests.sh

```diff
@@ -1,44 +1,56 @@
 #!/usr/bin/env bash
 # SPDX-License-Identifier: Apache-2.0
 #
+# This test script is laid out as follows:
+# - UTILITIES:  utility functions
+# - TESTS:      test functions
+# - SETUP:      environment setup steps
+# - MAIN:       test execution steps
+#
 # If you are running locally and calling the script multiple times you may want to run like this:
 #
 # TEST_DIR=/tmp/foo ./scripts/functional-tests.sh
 #
 # As soon as TEST_DIR is set to any non-empty value, the test directory will NOT be removed
 # This assumes you control your test directory TEST_DIR and its cleanup
 
-set -ex
-
-# shellcheck disable=SC2155
-export SCRIPTDIR=$(dirname "$0")
-# build a prompt string that includes the time, source file, line number, and function name
-export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
-
-if [ -n "$TEST_DIR" ]; then
-    echo "TEST_DIR is set to $TEST_DIR by the caller, will not remove it"
-    TEST_DIR_SET_BY_CALLER=1
-fi
-
-# Support overriding the test directory for local testing otherwise creates a temporary directory
-TEST_DIR=${TEST_DIR:-$(mktemp -d)}
-
-export TEST_DIR
-export PACKAGE_NAME='instructlab'  # name we use of the top-level package directories for CLI data
-
+#############
+# UTILITIES #
+#############
 
-# get the directories from the platformdirs library
-export CONFIG_DIR
-export DATA_DIR
-export CACHE_DIR
-
-
-# we define the path here to reference elsewhere, but the existence of this file
-# will be managed by the ilab CLI
-export ILAB_CONFIGDIR_LOCATION
-export ILAB_CONFIG_FILE
-export ILAB_DATA_DIR
+cleanup() {
+    set +e
```

Comment:
Do you know why this is here?

Reply:
Understood. Would be good to know what the specific reasons were. The trade-off is that we are now making sure each of the steps are running, but we have no feedback if they are actually working. In the future it would be good to isolate the isolate the specific failure conditions with an || true rather than using +e. Not blocking for this refactor.

Reply:
Agree 100% - @fredericlepied you implemented this at first - could you maybe shed some light on this?

Reply:
the idea of `set +e` is to to all the steps in the trap whatever happens as we need to clean everything.